src/lib/db.js

```diff
@@ -186,6 +165,34 @@ function createObservation (feature, nodeId, cb) {
   })
 }
 
+function getObservationById (id, cb) {
+  observationsDb.get(id, (err, doc) => {
+    cb(err, doc)
+  })
+}
+
+function getObservationTimestampStream (options, cb) {
+  var done, opts
+  if (typeof cb === 'undefined') {
+    done = options
+    opts = {}
+  } else {
+    done = cb
+    opts = options
+  }
+  observationsTimestampIndex.ready(() => done(null, observationsTimestampIndex.getDocumentStream(opts)))
+}
+
+function listSequentialObservations (cb) {
+  const ids = []
+  getObservationTimestampStream((err, stream) => {
+    if (err) cb(err)
+    stream.on('data', d => ids.push(d))
+    stream.on('end', () => cb(null, ids))
+    stream.on('error', error => cb(error))
```

Comment:
I don't think Readable streams guarantee that `'error'` will only be emitted once. I'd feel safer if we wrapped `cb` in a [`once`](https://www.npmjs.com/package/once).